src/execution/vm/bytecode_generator.cpp

```diff
@@ -1887,6 +1887,100 @@ void BytecodeGenerator::VisitBuiltinArithmeticCall(ast::CallExpr *call, ast::Bui
   GetExecutionResult()->SetDestination(dest);
 }
 
+void BytecodeGenerator::VisitBuiltinAtomicAndCall(ast::CallExpr *call) {
+  auto args = call->Arguments();
+  LocalVar dest = VisitExpressionForRValue(args[0]);
+  LocalVar val = VisitExpressionForRValue(args[1]);
+  LocalVar ret;
+  auto return_used = GetExecutionResult() != nullptr;
+  if (return_used) {
+    ret = GetExecutionResult()->GetOrCreateDestination(call->GetType());
+    GetExecutionResult()->SetDestination(ret);
+  } else {
+    ret = GetCurrentFunction()->NewLocal(call->GetType());
+  }
+
+  switch (args[1]->GetType()->GetSize()) {
```

Comment:
Switches for sizes seems odd for two reasons:
1. It isn't consistent with the rest of the style in this file.
2. It forces you to have a default clause despite type resolution having completed.

Use `if-then-else` with an assert in the last `else` clause.